crates/cheatcodes/src/utils.rs

```diff
@@ -181,3 +167,134 @@ impl Cheatcode for copyStorageCall {
+        Ok(DynSolValue::type_strategy(&DynSolType::Bool)
+            .new_tree(state.test_runner())
+            .unwrap()
+            .current()
+            .abi_encode())
+    }
+}
+
+impl Cheatcode for arbitraryBytesCall {
+    fn apply(&self, state: &mut Cheatcodes) -> Result {
+        let Self { len } = *self;
+        let mut val = DynSolValue::type_strategy(&DynSolType::Bytes)
+            .new_tree(state.test_runner())
+            .unwrap()
+            .current()
+            .as_bytes()
+            .unwrap_or_default()
+            .to_vec();
+        let required_len = len.to::<usize>();
+        let cur_len = val.len();
+        if cur_len > required_len {
+            // Slice to required length if random bytes length is lower than required length.
+            Ok(val[..required_len].abi_encode())
+        } else {
+            // Fill with zeroes if random bytes length is lower than required length.
+            val.extend(vec![0; required_len - cur_len]);
+            Ok(val.abi_encode())
+        }
+    }
+}
+
+/// Helper to generate an arbitrary `uint` value (with given bits or bounded if specified)
+/// from type strategy.
+fn arbitrary_uint(
+    state: &mut Cheatcodes,
+    bits: Option<U256>,
+    bounds: Option<(U256, U256)>,
+) -> Result {
+    if let Some(bits) = bits {
+        // Generate arbitrary with specified bits.
+        let no_bits = bits.to::<usize>();
+        ensure!(no_bits <= 256, "no of bits cannot exceed 256");
+        return Ok(DynSolValue::type_strategy(&DynSolType::Uint(no_bits))
+            .new_tree(state.test_runner())
+            .unwrap()
+            .current()
+            .abi_encode())
+    }
+
+    if let Some((min, max)) = bounds {
+        ensure!(min <= max, "min must be less than or equal to max");
+        // Generate arbitrary between range min..=max
+        let exclusive_modulo = max - min;
+        let mut random_number: U256 = state.test_runner().rng().gen();
+        if exclusive_modulo != U256::MAX {
+            let inclusive_modulo = exclusive_modulo + U256::from(1);
+            random_number %= inclusive_modulo;
+        }
+        random_number += min;
+        return Ok(random_number.abi_encode())
+    }
+
+    // Generate arbitrary `uint256` value.
+    Ok(DynSolValue::type_strategy(&DynSolType::Uint(256))
+        .new_tree(state.test_runner())
+        .unwrap()
+        .current()
+        .abi_encode())
+}
+
+/// Helper to generate an arbitrary `address` value from type strategy.
+fn arbitrary_address(state: &mut Cheatcodes) -> Result {
+    Ok(DynSolValue::type_strategy(&DynSolType::Address)
+        .new_tree(state.test_runner())
+        .unwrap()
+        .current()
+        .abi_encode())
+}
+
+/// Helper to generate an arbitrary `int` value (with given bits if specified) from type strategy.
+fn arbitrary_int(state: &mut Cheatcodes, bits: Option<U256>) -> Result {
+    let no_bits = bits.unwrap_or(U256::from(256)).to::<usize>();
```

Comment:
This can panic, move the `to()` to after the `ensure!`

Reply:
oops, added https://github.com/foundry-rs/foundry/pull/8882/commits/1c2e688c7e76bc79801e052a49239ea3061793ce + test